tests/test_utils.py

```diff
@@ -350,3 +351,15 @@ def test_instantiate_torch_optimizer_with_class(model_parameters):
     assert isinstance(optimizer, torch.optim.Adam)
     # init args gets overridden
     assert optimizer.param_groups[0]["lr"] == 0.02
+
+
+@pytest.mark.parametrize("input_path, expected", [
+    (Path("checkpoints/my_model"), Path("checkpoints/my_model")),                  # already prefixed
+    # the one below it to simulate what happens if the input is not a valid dir
+    (Path("data/my_model/file.txt"), Path("checkpoints/data/my_model/file.txt")),  # needs prefix
+    (Path("/data/my_model"), Path("/data/my_model")),                              # absolute path
+    (Path("./file.txt"), Path("checkpoints/./file.txt")),                          # current directory
+    (Path("../relative_path"), Path("checkpoints/../relative_path")),              # relative parent path
```

Comment:
This last case looks very unexpected. We probably shouldn't blindly add a prefix without resolving properly first. As a simple way to avoid this, my suggestion above for an existence check could help with this.